src/main/java/com/connexta/search/index/IndexManagerImpl.java

```diff
@@ -39,24 +41,25 @@ public void index(
     try {
       idAlreadyExists = crudRepository.existsById(productId);
     } catch (final Exception e) {
-      throw new IndexException("Unable to query index", e);
+      throw new IndexException(INTERNAL_SERVER_ERROR, "Unable to query index", e);
     }
     if (idAlreadyExists) {
-      throw new IndexException("Product already exists. Overriding is not supported");
+      throw new IndexException(
+          INTERNAL_SERVER_ERROR, "Product already exists. Overriding is not supported");
     }
 
     final Index index;
     try {
       index = new Index(productId, IOUtils.toString(inputStream, StandardCharsets.UTF_8));
     } catch (IOException e) {
-      throw new IndexException("Unable to convert InputStream to String", e);
+      throw new IndexException(INTERNAL_SERVER_ERROR, "Unable to convert InputStream to String", e);
```

Comment:
✏️ I think this case should return 400.

Reply:
Do we have enough evidence to conclude the request was in fact bad? Is there anything about our application that might mishandled the stream along the day?